src/Utilities/StdConstexprHelpers.hpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/* These functions are constexpr reimplementations of some standard library
+ * functions, designed to work with cpp17::array. Therefore, these functions
+ * are in the cpp17 namespace.
+ */
+
+#pragma once
+
+#include "Utilities/Gsl.hpp"
+
+namespace cpp17 {
+/* Reimplementation of std::accumulate that is constexpr
+ * Taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/numeric
+ */
+template <class ForwardIterator, class T>
+constexpr void iota(ForwardIterator first, ForwardIterator last,
+                    T value) noexcept {
+  for (; first != last; ++first, (void)++value) {
+    *first = value;
+  }
+}
+
+/* Reimplementation of std::accumulate that is constexpr
+ * Taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/numeric
+ */
+template <class InputIt, class T>
+constexpr T accumulate(InputIt first, InputIt last, T init) noexcept {
+  for (; first != last; ++first) {
+    init = init + *first;  // std::move since C++20
```

Comment:
I don't understand this inline comment here

Reply:
Copied and pasted from llvm. Init on RHS is std::move wrapped starting in c++20.